src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsallocations/config/WebClientUserEnhancementConfiguration.kt

```diff
@@ -64,53 +56,58 @@ class WebClientUserEnhancementConfiguration(
   }
 
   @Bean
-  @RequestScope
   fun workforceAllocationsToDeliusApiWebClientUserEnhancedAppScope(
-    clientRegistrationRepository: ClientRegistrationRepository,
+    clientRegistrationRepository: ReactiveClientRegistrationRepository,
     builder: WebClient.Builder
   ): WebClient {
-    return getOAuthWebClient(authorizedClientManagerUserEnhanced(clientRegistrationRepository), builder, workforceAllocationsToDeliusApiRootUri, "workforce-allocations-to-delius-api")
+    return getOAuthWebClient(authorizedClientManagerUserEnhanced(clientRegistrationRepository, builder), builder, workforceAllocationsToDeliusApiRootUri, "workforce-allocations-to-delius-api")
   }
 
   @Bean
   fun workforceAllocationsToDeliusApiClientUserEnhanced(@Qualifier("workforceAllocationsToDeliusApiWebClientUserEnhancedAppScope") webClient: WebClient): WorkforceAllocationsToDeliusApiClient {
     return WorkforceAllocationsToDeliusApiClient(webClient)
   }
 
-  private fun authorizedClientManagerUserEnhanced(clients: ClientRegistrationRepository?): OAuth2AuthorizedClientManager {
-    val service: OAuth2AuthorizedClientService = InMemoryOAuth2AuthorizedClientService(clients)
-    val manager = AuthorizedClientServiceOAuth2AuthorizedClientManager(clients, service)
+  private fun authorizedClientManagerUserEnhanced(clients: ReactiveClientRegistrationRepository?, builder: WebClient.Builder): ReactiveOAuth2AuthorizedClientManager {
+    val service: ReactiveOAuth2AuthorizedClientService = InMemoryReactiveOAuth2AuthorizedClientService(clients)
+    val manager = AuthorizedClientServiceReactiveOAuth2AuthorizedClientManager(clients, service)
+    val reactiveClientCredentialsTokenResponseClient = WebClientReactiveClientCredentialsTokenResponseClient()
 
-    val defaultClientCredentialsTokenResponseClient = DefaultClientCredentialsTokenResponseClient()
+    reactiveClientCredentialsTokenResponseClient.setWebClient(builder.filter(userEnhancedTokenRequestProcessor()).build())
```

Comment:
reactive equivalent of `setRequestEntityConverter` where you can modify the request before requesting a client creds oauth token